snaps/concepts/security-guidelines.md

```diff
@@ -0,0 +1,247 @@
+---
+description: Learn about best practices for creating secure and reliable Snaps.
+sidebar_position: 6
+---
+
+# Snaps security guidelines
+
+This page outlines essential principles for builders to develop secure and reliable Snaps.
+Use these guidelines when creating your Snap to ensure it is safe for users.
+
+## Permissions
+
+The following are guidelines for using the Snap [manifest file](../concepts/anatomy.md/#manifest-file)
+and its [permissions](../reference/permissions.md):
+
+- **Minimum permissions** - Follow the principle of least authority: only add the minimum
+  permissions needed by your Snap in the manifest file.
+  Before publishing your Snap, check the permissions again and remove any unused permissions.
+  You can validate your permissions using the [Snaps Simulator Manifest
+  Validator](https://metamask.github.io/snaps/snaps-simulator/staging/#/manifest).
+
+- **Minimum RPC access** - When adding the [`endowment:rpc`](../reference/permissions.md/#endowmentrpc)
+  permission for Snaps or dapps, ask yourself if both are necessary.
+  For example, if permission is granted to communicate with Snaps, it means other Snaps can call
+  your Snap's sensitive RPC methods.
+
+- **Minimum network access** - Only add the [`endowment:network-access`](../reference/permissions.md/#endowmentnetwork-access)
+  permission if it's absolutely necessary, such as when needing to communicate with a remote API
+  that is part of your Snap's functionality.
+  This to mitigate users' concern about sharing wallet usage to remote servers.
+  If your Snap needs this permission, inform the user before communicating with remote servers and
+  include a privacy policy in your Snap that explains how data is shared.
+
+## Transactions
+
+The following are guidelines for handling and signing transactions:
+
+- **Transaction details** - When handling transactions, present the user with a prompt displaying
+  all transaction details such as the receiver address, chain ID, network, and amount.
+  Show the domain name of the site that originated the transaction request, and the blockchain it
+  will be sent to.
+  The receiver address is crucial to prevent the user from sending funds to the wrong destination.
+
+- **Transparent signing** - Display the message to be signed by the user in the Snap confirmation flow.
+  Do not rely on the requesting site for this, as it can choose not to display it and let the
+  Snap silently sign the message.
```

Comment:
```suggestion
- **Transparent signing** - Display the message to be signed by the user in the Snap confirmation flow.
  Do not rely on the requesting site to display this message, as it might choose not to, leading to the Snap signing the message silently.
```